google/cloud/bigtable/internal/query_plan.cc

```diff
@@ -21,11 +21,30 @@ namespace cloud {
 namespace bigtable_internal {
 GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
 
+bool QueryPlan::IsExpired() const {
+  auto total_duration =
+      std::chrono::seconds(response_.valid_until().seconds()) +
+      std::chrono::nanoseconds(response_.valid_until().nanos());
+
+  auto valid_until = std::chrono::system_clock::time_point() + total_duration;
+  return valid_until <= std::chrono::system_clock::now();
+}
+
 std::string const& QueryPlan::prepared_query() const {
```

Comment:
For now let's change the return type to `StatusOr<std::string>` and return `kUnavailable` when IsExpired.

Reply:
Returning kUnavailable for now, thanks for the suggestion!